src/api/app/controllers/webui/comments_controller.rb

```diff
@@ -1,29 +1,35 @@
 class Webui::CommentsController < Webui::WebuiController
   before_action :require_login
+  before_action :set_commentable_type, only: :create
   before_action :find_commentable, only: :create
 
   def create
-    comment = @commented.comments.new(permitted_params)
-    User.session!.comments << comment
-    @commentable = comment.commentable
-
-    status = if comment.save
-               flash.now[:success] = 'Comment created successfully.'
-               :ok
-             else
-               flash.now[:error] = "Failed to create comment: #{comment.errors.full_messages.to_sentence}."
-               :unprocessable_entity
-             end
-
-    if Flipper.enabled?(:request_show_redesign, User.session) && comment.commentable_type == 'BsRequest'
-      render(partial: 'webui/comment/beta/comments_thread',
-             locals: { comment: comment.root, commentable: @commentable, level: 1 },
-             status: status)
+    if @commented.nil?
+      flash.now[:error] = "Failed to create comment: This #{@commentable_type.name.downcase} does not exist anymore."
+      render partial: 'layouts/webui/flash'
     else
```

Comment:
Introducing this `if - else` statement makes codeclimate warn with "Method create has a Cognitive Complexity of 9 (exceeds 8 allowed).". Consider using [`and return`](https://guides.rubyonrails.org/layouts_and_rendering.html#avoiding-double-render-errors) after `render partial`, so the `else` clause can be removed.

Reply:
Done